BUILDING.md

```diff
@@ -53,66 +29,169 @@ following packages are installed:
-Python module.
+OpenAssetIO consists of a core C++ project, as well as a set of C and
+Python bindings to that core libary. These modules are organized into a
+CMake project, and are all built together by default.
 
-In order to build and install the binary artifacts and Python sources,
-assuming the current working directory is at the repository root, run
+Assuming your build environment has the required dependencies,
+[(See Above)](#build-dependencies), OpenAssetIO can be built and installed by
+running :
 
 ```shell
 cmake -S . -B build
 cmake --build build
 cmake --install build
 ```
 
-The generated install tree can then be used directly (e.g. by extending
+> **Note**
+>
+> A common error to encounter is to find that CMake cannot resolve
+> its dependencies via [`find_package`](https://cmake.org/cmake/help/latest/command/find_package.html)
+>
+> A convenient way to solve this is to use the [`CMAKE_TOOLCHAIN_FILE`](https://cmake.org/cmake/help/v3.24/envvar/CMAKE_TOOLCHAIN_FILE.html)
+> environment variable to augment CMake's package search paths.
+>
+> For example, to use [`conan`](https://docs.conan.io/en/1.46/installation.html)
+> to resolve dependencies, (as we do during internal development) run
+> the following commands from the root of the repository prior to
+> building :
+>
+> ```shell
+> conan install -if .conan resources/build --build=missing
+> export CMAKE_TOOLCHAIN_FILE=`pwd`/.conan/conan_paths.cmake
+> ```
+>
+> (These commands assume a Linux host, translate as necessary).
+>
+> This installs OpenAssetIO's dependencies as described in
+> `/resources/build/conanfile.py`, and allows CMake to deduce the
+> locations via referencing the generated `conan_paths.cmake`.
+
+The modules are installed to `/build/dist`. This default location can
+be overridden by setting `CMAKE_INSTALL_PREFIX` prior to build.
+
+By default, OpenAssetIO builds as a shared library. This can be
+overriden by setting `BUILD_SHARED_LIBS` prior to build.
+If shared library builds are disabled, then the core library will
+be built and installed as a static library, and statically linked into
+the Python/C modules.
+
+In installing the python module, OpenAssetIO creates a valid python
+install tree, placed at `build/dist/{python-version}/site-packages`.
+This generated install tree can then be used directly (e.g. by extending
 `PYTHONPATH` and/or `LD_LIBRARY_PATH`), or as the source for a further
 packaging step.
 
-Note that the `openassetio` Python package will not function unless the
-binary artifacts are also made available to your Python environment.
+> **Note**
+>
+> When built this way, the `openassetio` Python package will not
+> function unless the binary artifacts are also made available to your
+> Python environment.
+>
+
+### Building via pip
+
+OpenAssetIO can alternately be built and installed via pip, which may be
+preferable to the exclusively python focused developer.
+Check out the repository, and from the root, run :
+
+```shell
+pip install .
```

Comment:
```suggestion
python -m pip install .
```

Reply:
I'm curious about this preference for `python -m pip`?  Is it just that it works for a venv without having to activate it?

Reply:
https://snarky.ca/why-you-should-use-python-m-pip/

Reply:
eg, on my Mac, some of my venvs don't have `pip` in (not looked into why yet), so activating them , uses the wrong python when you run `pip` but not `python -m pip`, so its universally safer.

Reply:
[d571a5a](https://github.com/OpenAssetIO/OpenAssetIO/pull/711/commits/d571a5aee9a1749f91b0c15284e6f4b5371f50d3)